src/core/meshcop/joiner.cpp

```diff
@@ -697,3 +673,37 @@ void Joiner::LogCertMessage(const char *aText, const Coap::Message &aMessage) co
 } // namespace ot
 
 #endif // OPENTHREAD_CONFIG_JOINER_ENABLE
+
+#if OPENTHREAD_CONFIG_JOINER_ENABLE || OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
+
+namespace ot {
+namespace MeshCoP {
+
+bool Joiner::IsPskdValid(const char *aPskd)
```

Comment:
Given that this method is used by both Commissioner and Joiner, should we move this to a more neutral location? For example, [src/core/meshcop/meshcop.hpp](https://github.com/openthread/openthread/blob/master/src/core/meshcop/meshcop.hpp).

Reply:
Good idea! will do.

Reply:
@jwhui Relocated to [meshcop.hpp](https://github.com/openthread/openthread/blob/master/src/core/meshcop/meshcop.hpp), please review.